### PR TITLE
Add limit for remove_negative and fix zoneKey error

### PR DESCRIPTION
**Pull request by systemcatch**
I used countryCode instead of zoneKey in #1191. Also added a limit to remove_negative as bigger negative numbers likely indicate something is wrong and requires attention.

### CODE REVIEW

File: parsers/lib/validation.py [hunk 1]
```diff
@@ -71,8 +71,8 @@ def validate(datapoint, **kwargs):
 
     if remove_negative:
         for key, val in generation.items():
-            if val is not None and val < 0.0:
-                print("{} returned negative value for {}".format(key,datapoint['countryCode']))
+            if val is not None and -5.0 < val < 0.0:
+                print("{} returned {} for {}, setting to None".format(key, val, datapoint['zoneKey']))
```
Comment (corradio) [inline parsers/lib/validation.py]: can we use `logger.warn` to log the warning?

Comment (systemcatch) [reply parsers/lib/validation.py]: sure I will do the same for the other prints, do I put logger=None in the function args?

**Comment by corradio** [comment]
if you us the logger and set logger=None we might get some errors if no
arguments is passed. Better force the logger argument?

**Comment by systemcatch** [comment]
ok so have it like def validate(datapoint, logger, **kwargs): then change the validate uses in each parser?

**Comment by corradio** [comment]
Yes indeed

**Comment by systemcatch** [comment]
When altering the validate line in IN-GJ to:
```data = validate(data, logger, required=['hydro'], floor=10.0)```

The following happens because logger=None is the default when run as main.
```
(EM-env) chris@ThinkPad:~/electricitymap$ python -m parsers.IN_GJ
Traceback (most recent call last):
  File "/usr/lib/python3.5/runpy.py", line 184, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/chris/electricitymap/parsers/IN_GJ.py", line 138, in <module>
    print(fetch_production('IN-GJ', session))
  File "/home/chris/electricitymap/parsers/IN_GJ.py", line 109, in fetch_production
    valid_data = validate(data, logger, remove_negative=True)
  File "/home/chris/electricitymap/parsers/lib/validation.py", line 77, in validate
    logger.warning("{} returned {}, setting to None".format(
AttributeError: 'NoneType' object has no attribute 'warning'
```
To get around this we could construct a temporary logger when running from the command line.
```
if __name__ == '__main__':
    logger = logging.getLogger(__name__)
    print(fetch_production())

**Comment by corradio** [comment]
Or maybe the temporary logger could be set as default argument?

**Comment by systemcatch** [comment]
Yes that is probably better as update_state.py does not run the functions as main so it'll fail.

**Comment by systemcatch** [comment]
@corradio should be good to go here, let me know if you want any changes.

**Pull request merged**